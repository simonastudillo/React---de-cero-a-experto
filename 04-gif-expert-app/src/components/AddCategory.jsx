import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  // Estados
  const [inputValue, setInputValue] = useState('');
  // Funciones
  // Permite el cambio de valor del input
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  // Envia el valor al parent
  const onSubmit = (e) => {
    e.preventDefault();
    // console.log('onSubmit');
    // Limpiamos el input
    const newInputValue = inputValue.trim();
    // Enviamos el valor al parent
    const isValid = onNewCategory(newInputValue);
    // validamos que el valor se ingreso correctamente
    if (!isValid) return;
    // Limpiamos el input en caso de que se ingrese correctamente
    setInputValue('');
  };
  // render
  return (
    <form onSubmit={onSubmit} aria-label="form-add-category">
      <input
        type="text"
        placeholder="Busca tus gifs favoritos"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}