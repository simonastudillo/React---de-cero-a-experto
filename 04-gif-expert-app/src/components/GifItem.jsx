import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}
// Tarea:
// 1. AÃ±adir PropTypes
//    - title: string obligatorio
//    - url: string obligatorio
// 2. Evaluar el snapchot

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}