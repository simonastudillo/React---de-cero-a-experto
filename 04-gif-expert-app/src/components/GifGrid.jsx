import { GifItem } from "./GifItem";
import { PropTypes } from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({ category, onRemoveCategory }) => {

  // const [gifs, setGifs] = useState([]);

  // useEffect(() => {
  //   getGifs(category)
  //     .then(gifs => setGifs(gifs));
  // }, []);

  // const awaitForGifs = async (category) => {
  //   const gifs = await getGifs(category);
  //   setGifs(gifs);
  // };

  // useEffect(() => {
  //   awaitForGifs(category);
  // }, []);

  // Custom hook 
  const { gifs, isLoading } = useFetchGifs(category);


  return (
    <>
      <h3>{category} <button onClick={() => onRemoveCategory(category)}>Eliminar</button></h3>
      {
        isLoading && (<h2>Cargando...</h2>)
      }
      <div className="card-grid">
        {gifs.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};