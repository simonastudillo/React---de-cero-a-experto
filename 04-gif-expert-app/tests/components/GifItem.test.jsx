import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Prueba de GifItem', () => {

  const title = 'Luffy';
  const url = 'https://media2.giphy.com/media/fZdzEHC8sMC0E/giphy.gif?cid=a3fd93602nim556ngligta53mbrcvcbaalff75ri4606r127&ep=v1_gifs_search&rid=giphy.gif&ct=g';

  test('debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe de mostrar el tÃ­tulo en el P', () => {
    render(<GifItem title={title} url={url} />)
    // const { innerHTML } = screen.getByRole('paragraph');
    // expect(innerHTML).toBe(title);
    expect(screen.getByText(title)).toBeTruthy();
  });

});