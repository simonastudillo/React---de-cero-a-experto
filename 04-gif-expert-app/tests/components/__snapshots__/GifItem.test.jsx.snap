// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Luffy"
      src="https://media2.giphy.com/media/fZdzEHC8sMC0E/giphy.gif?cid=a3fd93602nim556ngligta53mbrcvcbaalff75ri4606r127&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Luffy
    </p>
  </div>
</div>
`;
